pkg/providers/vsphere/session/session_vm_update.go

```diff
@@ -1046,6 +1046,15 @@ func (s *Session) UpdateVirtualMachine(
 		return err
 	}
 
+	// If VM is being paused, only try to update status once.
+	if paused := updateVMPaused(vmCtx, moVM); paused {
+		vmCtx.Logger.Info("Pausing Updating VirtualMachine")
+		if updateErr := vmlifecycle.UpdateStatus(vmCtx, s.K8sClient, vcVM, moVM); updateErr != nil {
+			vmCtx.Logger.Error(updateErr, "Updating VM status failed when VirtualMachine is paused")
+		}
+		return nil
+	}
```

Comment:
I would rename the `updateVMPaused` function to `isVMPaused` and simply return "admin/devops/both" or an empty string if the VM is not paused. Then, inside the `if` statement here, update the VM's labels accordingly.

Besides, I don't think the `UpdateStatus()` call is necessary here, as it goes through all VM's status fields. Instead, you could simply call `conditions.MarkFalse/MarkTrue` based on the returned value from the above function.

Reply:
I think we'd still want to keep the VM Status up to date. The "only try to update status once" comment doesn't make sense to me. 

I've got some larger changes planned to this area but I'd like to keep the single call to `UpdateStatus()` at the end of this function. Something like below would work and is still pretty readable. 

```
if !paused {
   switch vmCtx.VM.Spec.PowerState {
```

Reply:
renamed the `updateVMPaused` function to `isVMPaused`. But I still decide to do label update inside the func. Returning bool for `isVMPaused` is more straightforward and can be used in later.